src/unlinkcommand.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Command from '../core/command/command.js';
+import Range from '../engine/model/range.js';
+
+/**
+ * The unlink command. It is used by the {@link Link.Link link feature}.
+ *
+ * @memberOf link
+ * @extends core.command.Command
+ */
+export default class LinkCommand extends Command {
+	/**
+	 * Executes the command.
+	 *
+	 * When selection is collapsed then whole link will be unlinked. {@link engine.model.TreeWalker TreeWalker} is
+	 * looking for link bounds going forward and backward from the caret position.
+	 *
+	 * If selection is non-collapsed then only selected range will be unlinked.
+	 *
+	 * @protected
+	 */
+	_doExecute() {
+		const document = this.editor.document;
+		const selection = document.selection;
+		const ranges = selection.getRanges();
+		let rangesToUpdate = [];
+
+		document.enqueueChanges( () => {
+			// When selection is collapsed we have to find link bounds.
+			if ( selection.isCollapsed ) {
+				// Loop through each selection.
+				for ( let range of ranges ) {
+					// Get searching area.
```

Comment:
Once again, looping through collapsed selection. This time it makes a little more sense, because theoretically, you could have a few collapsed ranges in different links in editor and you might want to remove them all at once. Unfortunately, this won't work, because `Selection` is collapsed only when it has exactly one collapsed range.